Source/WebCore/layout/formattingContexts/inline/InlineFormattingContext.cpp

```diff
@@ -497,6 +497,13 @@ bool InlineFormattingContext::rebuildInlineItemListIfNeeded(const InlineDamage*
     return true;
 }
 
+std::optional<size_t> InlineFormattingContext::numberOfVisibleLinesAllowed() const
+{
+    if (auto lineClamp = layoutState().parentBlockLayoutState().lineClamp())
+        return lineClamp->maximumLineCount > lineClamp->currentLineCount ? lineClamp->maximumLineCount - lineClamp->currentLineCount : 0;
+    return { };
+}
```

Comment:
any particular reason why this has to be on InlineFormattingContext now?

Reply:
It's just that this is already calculated at `InlineFormattingContext::createDisplayContentForInlineContent`, so I wanted to reuse it (Which I forgot to do in this patch).

Reply:
please don't. InlineFormattingContext's API should stay focused. if you are interested in sharing code, move it to a helper function to the util class but I don't think it's worth the effort.